File: Homework3/Question2/src/Animal.java
public class Animal {
    private String color;

    public Animal(String color) {
        this.color = color;
    }

    public String getColor() {
        return color;
    }

    public boolean isEdible() {
        System.out.println("Do not eat this!");
        return false;
    }

    public static void testAnimal() {
        Animal animal = new Animal("green");
        System.out.println(animal.getColor() == "green");
        System.out.println(animal.isEdible() == false);
    }
}

File: Homework3/Animal.kt
open class Animal {
    private val color: String

    constructor(color: String) {
        this.color = color
    }

    fun getColor(): String {
        return color
    }

    open fun isEdible(): Boolean {
        println("Do not eat this!")
        return false
    }

    companion object {
        fun testAnimal() {
            val animal = Animal("green")
            println(animal.getColor() == "green")
            println(animal.isEdible() == false)
        }
    }
}